intra_process_demo/src/image_pipeline/camera_node.cpp

```diff
@@ -21,7 +21,16 @@
 int main(int argc, char ** argv)
 {
   rclcpp::init(argc, argv);
-  auto camera_node = std::make_shared<CameraNode>("image");
-  rclcpp::spin(camera_node);
-  return 0;
+  try
+  {
+    auto camera_node = std::make_shared<CameraNode>("image");
+    rclcpp::spin(camera_node);
+    return 0;
+  }
+  catch( const std::exception& e )
+  {
+    fprintf(stderr, "Failed to open camera node. Exiting..\n");
```

Comment:
So long as you're catching the exception, you might as well include the `.what()` of the exception in the message. Otherwise, I'd suggest catching the exact error that is being thrown.